dashboard-improvements/README.md

```diff
@@ -0,0 +1,151 @@
+# Improving dashboard
+
+## Main points
+
+- Informing users about results of SRPM, COPR and Koji builds or Testing Farm
+- Replace service "Build Result" webpage with dashboard
+  - Deprecation of webpage in service
+- Database connection for dashboard
+  - Consider sensitivity of information provided (even with read-only access)
+  - Consider customizing API to the needs of dashboard
+- Consider possible improvements related to either of the points
+
+## Related issues
+
+- [Have a separate page for Testing farm](https://github.com/packit/packit-service/issues/506)
+  - Seems to be at least partially fixed: https://dashboard.stg.packit.dev/results/testing-farm/5603
+  - I guess we need to interlink all related jobs (SRPM, COPR and TF)
+- [IDs for COPR builds in API endpoint (bug)](https://github.com/packit/packit-service/issues/982)
+  - Does not affect usage of COPR build result page so far (is not used), or is the
+    reason for only one chroot result being displayed?
+
+## Current state
+
+- [Already existing TODO list](https://github.com/packit/dashboard/blob/main/TODO.md)
+- We have empty homepage
+  - I can see a potential in showing some statistics, could be helpful to us
+    and also to users.
+  - We could list recently finished jobs or queue.
+- We have implemented result pages
+  - COPR builds:bug mentioned above
+  - Minimal info for testing farm
+- For COPR builds we link directly COPR logs on the dashboard (recent COPR builds),
+  we should probably address our own result page before the COPR one.
+- Navigation:
+  - From each of those we can get back to the branch or the PR
+  - SRPM results -> {can see logs, download SRPM if available}
+  - COPR results -> {SRPM results, COPR logs, compressed build logs from COPR}
+  - TF results -> dead end
+
+## Removal of the _build result_ webpage
+
+0. Fix COPR build bug mentioned above, so that we can see all chroots.
+1. Set dashboard link for all new commit statuses.
+2. Set 2-week long (conveniently matching sprint) grace period for build result
+   page in service.
+
+   _Argument_: 2 weeks is enough for populating commit statuses with links to dashboard
+   and in majority of cases, old links are mainly used for debugging purposes on
+   our side.
+
+3. Remove build result page from service.
+
+   _Alternatively_: Also redirect all requests from service to dashboard in step 1
+   and keep live for longer period of time.
```

Comment:
How much time do you think this will take?

Reply:
**Implementation?**
Dumping the HTML and adding redirect to it, 1-2 points maybe?

**How long do we want to keep it?**
That is entirely up to us, it is quite interchangeable, e.g.:
`https://stg.packit.dev/srpm-build/3977/logs ~> https://dashboard.stg.packit.dev/results/srpm-builds/3977`

As I said few lines above that, the only problem with switch are old links in statuses... And I'm not sure if anyone checks out the old statuses, unless there are problems present? (In that case it still should not be a problem, since we know about the change and can have a look at it on the dashboard or even API/postgres.

I gave the guess of 2 weeks which seems more than enough for smaller PRs, regarding bigger PRs I would expect the statuses to be replaced.

Reply:
Yes, the redirect itself is easy. So if the implementation is no that hard, I would go with this one.

Reply:
agreed with 2/4 weeks for statuses

we could even start with some warning and hint about the work